dev/WindowsAppRuntime_BootstrapDLL/MddBootstrap.cpp

```diff
@@ -815,3 +876,129 @@ CLSID GetClsid(const winrt::Windows::ApplicationModel::AppExtensions::AppExtensi
+
+    return S_OK;
+}
+CATCH_RETURN()
+
+HRESULT MddBootstrapInitialize_ShowUI(
+    UINT32 majorMinorVersion,
+    PCWSTR versionTag,
+    PACKAGE_VERSION minVersion)
+{
+    //TODO: Show this if ERROR_NO_MATCH vs simpler MessageBox("ERROR 0xN in Bootstrapper initialize", MB_OK) ?
+
+    // Get the message caption
+    PCWSTR caption{};
+    wil::unique_cotaskmem_string captionString;
+    WCHAR captionOnError[100]{};
+    try
+    {
+        PCWSTR executable{};
+        wil::unique_cotaskmem_string module;
+        auto hr{ LOG_IF_FAILED(wil::GetModuleFileNameW(nullptr, module)) };
+        if (SUCCEEDED(hr))
+        {
+            auto delimiter{ wcsrchr(module.get(), L'\\') };
+            if (delimiter)
+            {
+                executable = delimiter + 1;
+            }
+            else
+            {
+                executable = module.get();
+            }
+            PCWSTR captionSuffix{ L"This application could not be started" };
+            captionString = wil::str_printf<wil::unique_cotaskmem_string>(L"%s - %s", executable, captionSuffix);
+            caption = captionString.get();
+        }
+    }
+    catch (...)
+    {
+    }
+    if (!caption)
+    {
+        LOG_IF_FAILED(StringCchPrintfW(captionOnError, ARRAYSIZE(captionOnError),
+                                       L"<Process %d> - This application could not be started",
+                                       GetCurrentProcessId()));
+        caption = captionOnError;
+    }
+
+    // Get the message body
+    WCHAR text[1024]{};
+    PCWSTR textFormat{ L"This application requires the Windows App Runtime\n"
+                       L"    Version %hu.%hu%s\n"
+                       L"    (MSIX packages version >= %hu.%hu.%hu.%hu)\n"
+                       L"\n"
+                       L"Do you want to install a compatible Windows App Runtime now?"
+                     };
+    const UINT16 majorVersion{ HIWORD(majorMinorVersion) };
+    const UINT16 minorVersion{ LOWORD(majorMinorVersion) };
+    WCHAR formattedVersionTag[64]{};
+    if (versionTag && (versionTag[0] != L'\0'))
+    {
+        FAIL_FAST_IF_FAILED(StringCchPrintfW(formattedVersionTag, ARRAYSIZE(formattedVersionTag), L"-%s", versionTag));
+    }
+    FAIL_FAST_IF_FAILED(StringCchPrintfW(text, ARRAYSIZE(text), textFormat,
+                                         majorVersion, minorVersion, formattedVersionTag,
+                                         minVersion.Major, minVersion.Minor, minVersion.Build, minVersion.Revision));
+
+    // Show the prompt
+    const auto yesno{ MessageBoxW(nullptr, text, caption, MB_YESNO | MB_ICONERROR) };
+    if (yesno == IDYES)
+    {
+        SHELLEXECUTEINFOW sei{};
+        sei.cbSize = sizeof(sei);
+        sei.lpVerb = L"open";
+        sei.lpFile = L"https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/downloads";
+        //TODO:Replace with https://aka.ms/windowsappsdk/<major>.<minor>/latest/windowsappruntimeinstall-<architecture>.exe
```

Comment:
could create an aka.ms and pass version info as query params for now, then implement url support later

Reply:
We've got the URLs I just need to lookup the exact syntax. The TODO .../latest/... URL is what we ~should~ will invoke ;-)